services/download/src/main/java/app/coronawarn/server/services/download/FederationGatewayDownloadService.java

```diff
@@ -82,6 +82,31 @@ public BatchDownloadResponse downloadBatch(String batchTag, LocalDate date) thro
     }
   }
 
+  /**
+   * Audit the batch from the EFGS for the given date.
+   *
+   * @param batchTag The batchTag of the batch that should be audited.
+   * @param date     The date for which the batch should be audited.
+   */
+  public void auditBatch(String batchTag, LocalDate date) {
+    try {
+      ResponseEntity<String> auditInformation = federationGatewayClient
+          .getAuditInformation(date.format(ISO_LOCAL_DATE), batchTag);
+      logger.info("Retrieved response from EFGS:{}", auditInformation);
+    } catch (FeignException.BadRequest | FeignException.Forbidden | FeignException.NotAcceptable
+        | FeignException.Gone | FeignException.NotFound clientError) {
+      logger.error("Auditing batch " + batchTag + " for date " + date.format(ISO_LOCAL_DATE)
+          + " failed due to: " + clientError.getMessage());
+      throw new BatchAuditException("Downloading batch " + batchTag + " for date " + date + " failed due to: "
+          + clientError.getMessage());
```

Comment:
What about passing the `clientError` as cause to the `BatchAuditException` to keep the stack trace?